src/python/pants/backend/project_info/peek.py

```diff
@@ -162,21 +193,88 @@ def describe_ruleset(ruleset: DependencyRuleSet | None) -> tuple[str, ...] | Non
     return ruleset.peek()
 
 
+async def _create_target_alias_to_goals_map() -> dict[str, tuple[str, ...]]:
+    """Returns a mapping from a target alias to the goals that can operate on that target.
+
+    For instance, `pex_binary` would map to `("run", "package")`.
+
+    :return: A mapping from a target alias to the goals that can operate on that target.
+    """
+    # TODO: This is manually curated for now - do we want to show all possible goals?
```

Comment:
Anyone know if this is something that exists somewhere in the codebase?

Reply:
I don't think we have this elsewhere.. 
There is one "loose" mapping of goal -> field_set, that is not used extensively, one case can be seen here: 
https://github.com/pantsbuild/pants/blob/842c566b31b3f9ee3a06640a339d3d7f93508c09/src/python/pants/core/goals/fix.py#L195-L197

Reply:
I made a comment about this - it's indirect, at best. 

I don't want to tackle coupling these right now, because it seems straight-forward for a bunch of goals, but it might be a lot messier for others (linters)